### ISSUE TITLE
 Dataset lambda encryption settings

### ISSUE THREAD

**Issue by manjulaK** [data-dot-all/dataall#434]
### Describe the bug

Check encryption settings for Lambda environmental variable, all the lambdas used in data.all that use environment variables should use a KMS key to encrypt the environment variable. This is a a security best practice. Currently this is not true with data.all lambdas. Please fix this .

### How to Reproduce

```
*P.S. Please do not attach files as it's considered a security risk. Add code snippets directly in the message body as much as possible.*
```

### Expected behavior
all the lambdas used in data.all that use environment variables should use a KMS key to encrypt the environment variable

### OS

All

### Python version

3.1

### AWS data.all version

v1.3,v1.4,v1.5

**Comment by tvancasteren** [comment]
@manjulaK @dlpzx I'd like to add that Lambda environment variables by default are encrypted using an AWS-managed KMS key. From my review in the accounts, this also applies to data.all. Optionally, a KMS Customer Managed Key can be configured instead of AWS Managed.
https://docs.aws.amazon.com/lambda/latest/dg/security-dataprotection.html#security-privacy-atrest

Following this info, it might be helpful to re-evaluate the need for this issue.

**Comment by manjulaK** [comment]
hi @tvancasteren  thank you for looking into this. You are right all the env variables used in lambda are encrypted with aws managed but can there be a provision for customer managed key to be provided for all lambdas? This can be considered for an enhancement?

**Comment by dlpzx** [comment]
Hi @manjulaK, @tvancasteren  thank you for looking into the issue. We are reviewing the documentation and as @tvancasteren pointed out, it does not seem to be strictly necessary to implement this feature. But maybe you have other requirements where CMK keys are needed. @manjulaK can you provide more details on the necessity of this issue?

**Comment by dlpzx** [comment]
This has been fixed in #1322
